### PR TITLE
Add Google Tag Mgr plugin and its configuration

### PR DESCRIPTION
**Pull request by aileenrose**
Note:  currently there no active tags i.e. Google Analytics or Cookiebot configured for this domain yet.   Once the tag is installed,  I will test on Netlify staging site.

Once tested, the next step will be to migrate Analytics urchin code into GTM, and configure cookiebot.

### CODE REVIEW

**Comment by aileenrose** [comment]
No reviewers yet, need to confirm setting and configuration on staging site.

File: website/src/pages/privacy/cookie-policy.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import Layout from '../../components/Layout'
+export default class CookiesPage extends React.Component<any, any> {
+    constructor(props: any) {
+        super(props)
+        this.state = {
+            cookies: [],
+        }
+    }
+
+    public componentDidMount(): void {
```
Comment (ryan-blunden) [inline website/src/pages/privacy/cookie-policy.tsx]: Can we address this hack with CSS?

File: website/src/pages/privacy/cookie-policy.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    public render(): JSX.Element | null {
+        const desc =
+            'Sourcegraph cookie policy'
+        return (
+            <Layout location={this.props.location}
+                meta={{
+                    title:
+                        'Sourcegraph Cookie Policy',
```
Comment (ryan-blunden) [inline website/src/pages/privacy/cookie-policy.tsx]:
```suggestion
                        'Sourcegraph cookie policy',
```

Prefer sentence case

**Comment by ryan-blunden** [comment]
Few issues here I think are important to address:

## The consent form disappears upon page scroll and does not re-appear.

ScreenFlow

## It needs styling and content work

- It looks quite ugly and doesn't integrate with the style of the site.
- It's very wordy compared with other sites I've seen.

Screen Shot 2020-02-05 at 9 20 43 AM

## It's scary and complicated

The existence of the **Do Not Sell My Personal Information** implies we might (and perhaps already have been) selling people's private information which is obviously not true. I don't think this box needs to be there and content explicitly saying that "we don't share your personal details with third parties" should be sufficient.

Also, do we really need **details** dropdown with the tabs showing every type of cookie collected? I haven't seen this on any other site.

**Comment by aileenrose** [comment]
GitLab uses the default settings of Cookiebot.
Screen Shot 2020-02-13 at 3 54 01 PM

I'll continue to tweak the style.  I welcome your feedback.

> ## It's scary and complicated
> The existence of the **Do Not Sell My Personal Information** implies we might 

This is a CCPA requirement.

> Also, do we really need **details** dropdown with the tabs showing every type of cookie collected? I haven't seen this on any other site.

Details I can hide, as long as we have the link to the Cookie Policy which details all the cookies.

File: docs/privacy.md [hunk 3]
```diff
@@ -52,7 +52,7 @@ The only personal information collected is the email address of the initial Sour
...
 
-- **Cookies**. For instance, when you visit our Website, we may send one or more cookies — a small text file containing a string of alphanumeric characters — to your computer that uniquely identifies your browser and lets us help you log in faster and enhance your navigation through the site. A cookie may also convey information to us about how you use the Website (e.g., the pages you view, the links you click, how frequently you access the Website, and other actions you take on the Website), and allow us to track your usage of the Website over time.
+- **Cookies**. For instance, when you visit our Website, we may send one or more cookies — a small text file containing a string of alphanumeric characters — to your computer that uniquely identifies your browser and lets us help you log in faster and enhance your navigation through the site. A cookie may also convey information to us about how you use the Website (e.g., the pages you view, the links you click, how frequently you access the Website, and other actions you take on the Website), and allow us to track your usage of the Website over time.
```
Comment (ryan-blunden) [inline docs/privacy.md]:
```suggestion
- **Cookies**. When visiting our website, we may send one or more cookies — a small text file containing a string of alphanumeric characters — to your computer that uniquely identifies your browser and lets us help you log in faster and enhance your navigation through the site. A cookie may also convey information to us about how you use the Website (e.g., the pages you view, the links you click, how frequently you access the Website, and other actions you take on the Website), and allow us to track your usage of the Website over time.  
```

File: website/src/css/components/_Cookiebot.scss [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+#CybotCookiebotDialogBodyContentTitle,
+#CybotCookiebotDialogBodyButtons a {
+    font-family: Circular!important;
+}
+#CybotCookiebotDialog {
+    opacity: .90 !important;
+}
+#CybotCookiebotDialogBodyContent,
+#CybotCookiebotDialogBodyContentTitle {
+    font-weight:300!important;
```
Comment (ryan-blunden) [inline website/src/css/components/_Cookiebot.scss]:
```suggestion
    font-size: 1.2rem !important;
    font-weight:300 !important;
```

File: website/src/css/components/_Cookiebot.scss [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+}
+#CybotCookiebotDialogBody {
+    max-width: 1000px!important;
+}
```
Comment (ryan-blunden) [inline website/src/css/components/_Cookiebot.scss]:
```suggestion
}
#CybotCookiebotDialogBodyContentText {
    font-size: 0.9rem !important;
}
```

The text at the current size is difficult to read.

File: website/src/pages/privacy/cookie-policy.tsx (hunk 1 above)
Comment (ryan-blunden) [reply website/src/pages/privacy/cookie-policy.tsx]: @aileenrose Why is this needed? Can it be removed?

Comment (aileenrose) [reply website/src/pages/privacy/cookie-policy.tsx]: I'm not sure to do that.

Comment (aileenrose) [reply website/src/pages/privacy/cookie-policy.tsx]: removed.

Comment (aileenrose) [reply website/src/pages/privacy/cookie-policy.tsx]: wasn't clear about the hack with CSS you were referring to in the previous comment.

**Pull request merged**